pkg/controller/gitopsservice/kam.go

```diff
@@ -0,0 +1,215 @@
+package gitopsservice
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	console "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	pipelinesv1alpha1 "github.com/redhat-developer/gitops-operator/pkg/apis/pipelines/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const cliName = "kam"
+const cliLongName = "GitOps Application Manager"
+const cliImage = "quay.io/redhat-developer/kam:1d0ef5"
+const cliImageEnvName = "KAM_IMAGE"
+
+func newDeploymentForCLI() *appsv1.Deployment {
+	image := os.Getenv(cliImageEnvName)
+	if image == "" {
+		image = cliImage
+	}
+	podSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:  cliName,
+				Image: image,
+				Ports: []corev1.ContainerPort{
+					{
+						Name:          "http",
+						Protocol:      corev1.ProtocolTCP,
+						ContainerPort: port, // should come from flag
+					},
+				},
+			},
+		},
+	}
+
+	template := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"app": cliName,
+			},
+		},
+		Spec: podSpec,
+	}
+
+	var replicas int32 = 1
+	deploymentSpec := appsv1.DeploymentSpec{
+		Replicas: &replicas,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app": cliName,
+			},
+		},
+		Template: template,
+	}
+
+	deploymentObj := &appsv1.Deployment{
+		ObjectMeta: objectMeta(cliName, serviceNamespace),
+		Spec:       deploymentSpec,
+	}
+
+	return deploymentObj
+}
+
+func newServiceForCLI() *corev1.Service {
+
+	spec := corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Name:       "tcp-8080",
+				Port:       port,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(int(port)),
+			},
+			{
+				Name:       "tcp-8443",
+				Port:       portSSL,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(int(8443)),
+			},
```

Comment:
Shouldn't this be using portSSL? (which might be better named portTLS)

Reply:
Fixed and renamed 🚌